src/homework_two/index.ts

```diff
@@ -0,0 +1,308 @@
+    this._groups.push(group);
+  }
+
+  removeGroup(groupName: string): void {
+    this._groups = this._groups.filter(group => group.name !== groupName);
+  }
+}
+
+class Group implements IGroup {
+  // implement getters for fields and 'add/remove student' and 'set status' methods
+
+  _area: string;
+  _status: GroupStatusEnum;
+  _students: IStudent[] = []; // Modify the array so that it has a valid toSorted method*
+  _name: string;
+  _directionName: string;
+  _levelName: string;
+
+  constructor({ area, directionName, levelName, status }: TGroupContructor) {
+    this._directionName = directionName;
+    this._levelName = levelName;
+    this._area = area;
+    this._status = status;
+    this._name = `${directionName}-${area}-${levelName}`;
+  }
+
+  get area(): string {
+    return this._area;
+  }
+
+  get status(): GroupStatusEnum {
+    return this._status;
+  }
+
+  get directionName(): string {
+    return this._directionName;
+  }
+
+  get levelName(): string {
+    return this._levelName;
+  }
+
+  set status(newStatus: GroupStatusEnum) {
+    this._status = newStatus;
+  }
+
+  get students(): IStudent[] {
+    return this._students;
+  }
+
+  get name(): string {
+    return this._name;
+  }
+
+  showPerformance(): IStudent[] {
+    const sortedStudents: IStudent[] = this._students.toSorted(
+      (a, b) => b.getPerformanceRating() - a.getPerformanceRating()
+    );
+    return sortedStudents;
+  }
+
+  addStudent(newStudent: TNewStudent): void {
+    const student: IStudent = new Student(newStudent);
+    this._students.push(student);
+  }
+
+  removeStudent({ fullName, birthYear }: IStudent): void {
+    this._students = this._students.filter(
+      (student: IStudent) => fullName !== student.fullName && birthYear !== student.birthYear
+    );
+  }
+}
+
+class Student implements IStudent {
+  // implement 'set grade' and 'set visit' methods
+
+  _firstName: string;
+  _lastName: string;
+  _birthYear: number;
+
+  _grades: IStudentGrade[] = []; // workName: mark
+  _visits: IStudentVisit[] = []; // lesson: present
+
+  constructor({ birthYear, firstName, lastName }: TNewStudent) {
+    this._firstName = firstName;
+    this._lastName = lastName;
+    this._birthYear = birthYear;
+  }
+
+  get fullName(): string {
+    return `${this._lastName} ${this._firstName}`;
+  }
+
+  set fullName(value: string) {
+    [this._lastName, this._firstName] = value.split(' ');
+  }
+
+  get age(): number {
+    return new Date().getFullYear() - this._birthYear;
+  }
+
+  get firstName(): string {
+    return this._firstName;
+  }
+
+  get lastName(): string {
+    return this._lastName;
+  }
+
+  get birthYear(): number {
+    return this._birthYear;
+  }
+
+  get grades(): IStudentGrade[] {
+    return this._grades;
+  }
+
+  set grade(newGrade: IStudentGrade) {
+    this._grades.push(newGrade);
+  }
+
+  get visits(): IStudentVisit[] {
+    return this._visits;
+  }
+
+  set visit(newVisit: IStudentVisit) {
```

Comment:
The same. It's a mistake in task. You did everything right. But setter should be named `visits`.